prefect_azure/container_instance.py

```diff
@@ -0,0 +1,707 @@
+        description=(
+            "The amount of memory in gigabytes to provide to the ACI task. Valid "
+            "amounts are specified in the Azure documentation. If not provided, a "
+            f"default value of  {ACI_DEFAULT_MEMORY} will be used unless present "
+            "on the task definition."
+        ),
+    )
+    subnet_ids: List[str] = Field(
+        default=None,
+        description="A list of Azure subnet IDs the container should be connected to.",
+    )
+    stream_output: bool = Field(
+        default=None,
+        description=(
+            "If `True`, logs will be streamed from the Prefect container to the local "
+            "console."
+        ),
+    )
+    env: Dict[str, Optional[str]] = Field(
+        title="Environment Variables",
+        default_factory=dict,
+        description=(
+            "Environment variables to provide to the task run. These variables are set "
+            "on the Prefect container at task runtime. These will not be set on the "
+            "task definition."
+        ),
+    )
+    # Execution settings
+    task_start_timeout_seconds: int = Field(
+        default=240,
+        description=(
+            "The amount of time to watch for the start of the ACI container. "
+            "before marking it as failed."
+        ),
+    )
+    task_watch_poll_interval: float = Field(
+        default=5.0,
+        description=(
+            "The amount of time to wait between Azure API calls while monitoring the "
+            "state of an Azure Container Instances task."
+        ),
+    )
+
+    @validator("command")
+    def validate_command(cls, command: List[str]):
+        """
+        A Pydantic validator that ensures a command is available to run
+        in an Azure container instance.
+
+        Args:
+            command: The command to validate.
+
+        Returns:
+            A command list if present, otherwise the base flow run command.
+        """
+
+        if not command:
+            return cls._base_flow_run_command()
+        return command
+
+    @sync_compatible
+    async def run(
+        self, task_status: Optional[TaskStatus] = None
+    ) -> ContainerInstanceJobResult:
+        """
+        Runs the configured task using an ACI container.
+
+        Args:
+            task_status: An optional `TaskStatus` to update when the container starts.
+
+        Returns:
+            An `ContainerInstanceJobResult` with the container's exit code.
+        """
+
+        run_start_time = datetime.datetime.now(datetime.timezone.utc)
+        token_credential = self._create_credential()
+        aci_client = self._create_container_client(token_credential)
+        container = self._configure_container()
+        container_group = self._configure_container_group(token_credential, container)
+        created_container_group = None
+
+        self.logger.info(
+            f"{self._log_prefix}: Preparing to run command {' '.join(self.command)!r} "
```

Comment:
The default command won't, and I suppose someone _could_ include sensitive information in the command array if they wanted to, but it would be an odd place to put it. I'm happy to exclude command here if needed, but it's worth noting I took this log call from `ECSTask`: https://github.com/PrefectHQ/prefect-aws/blob/8af9da3cc2bcd8bc3db1e210311a77266839558a/prefect_aws/ecs.py#L464